### ISSUE TITLE
Build fails on Windows10 x86_64 Meteor 1.4.4.1

### ISSUE THREAD

**Issue by ExcelNet-Public** [meteor/meteor#8614]
Using the same build / deploy scripts for the past year now produces the following error:

```
>meteor build ../build --architecture os.linux.x86_64
node.js:489
        callback();
        ^

Error: ENOTEMPTY: directory not empty, rmdir 'C:\Users\myuser\meteor\excel.net\.meteor\local\build_tar\bundle\programs\server\npm\node_modules\meteor\cosmos_browserify\node_modules\browserify\test\noparse\node_modules\robot'
    at Error (native)
    at Object.fs.rmdirSync (fs.js:758:18)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:332:11)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at Function.rimrafSync [as sync] (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at Object.rm_recursive (C:\tools\fs\files.js:303:12)
    at buildCommand (C:\tools\cli\commands.js:1059:9)
    at Command.func (C:\tools\cli\commands.js:824:12)
    at C:\tools\cli\main.js:1483:23
```

Have tried completely removing the _build-tar_ folder and starting over but it results in the same issues being generated.

**Comment by ExcelNet-Public** [comment]
Using the same build / deploy scripts for the past year now produces the following error:

```
>meteor build ../build --architecture os.linux.x86_64
node.js:489
        callback();
        ^

Error: ENOTEMPTY: directory not empty, rmdir 'C:\Users\myuser\meteor\excel.net\.meteor\local\build_tar\bundle\programs\server\npm\node_modules\meteor\cosmos_browserify\node_modules\browserify\test\noparse\node_modules\robot'
    at Error (native)
    at Object.fs.rmdirSync (fs.js:758:18)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:332:11)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at rimrafSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:330:5
    at Array.forEach (native)
    at rmkidsSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:329:26)
    at rmdirSync (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:322:7)
    at Function.rimrafSync [as sync] (C:\Users\myuser\AppData\Local\.meteor\packages\meteor-tool\1.4.4_1\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9)
    at Object.rm_recursive (C:\tools\fs\files.js:303:12)
    at buildCommand (C:\tools\cli\commands.js:1059:9)
    at Command.func (C:\tools\cli\commands.js:824:12)
    at C:\tools\cli\main.js:1483:23
```

Have tried completely removing the _build-tar_ folder and starting over but it results in the same issues being generated.

**Comment by ExcelNet-Public** [comment]
As an observation to this issue, which only seems to be happening with larger projects it seems that the point of failure is "random".  Cleaning up and running multiple times will yield different locations for where the build fails.

Second time:
`Error: ENOTEMPTY: directory not empty, rmdir 'C:\Users\myuser\meteor\excel.net\.meteor\local\build_tar\bundle\programs\server\npm\node_modules\meteor\ddp-client\node_modules\faye-websocket\node_modules\websocket-driver\lib'`

Third time:
`Error: ENOTEMPTY: directory not empty, rmdir 'C:\Users\myuser\meteor\excel.net\.meteor\local\build_tar\bundle-garbage-1esv2q\programs\server\npm\node_modules\meteor\http\node_modules\sshpk\node_modules'
`

**Comment by Mklueh** [comment]
Same here. Happens out of nowhere with different dependencies

**Comment by hwillson** [comment]
This appears to be related to #8589 - can you try the steps outlined in https://github.com/meteor/meteor/issues/8589#issuecomment-293878808 and report back?

**Comment by dmarkrollins** [comment]
I was having this same issue symptom which I posted on on the [forum](https://forums.meteor.com/t/enotempty-error-running-app-on-windows-10/35793/4).

I think the source of my issue is slightly different but the end result is the same. 

In my case the files names referenced in package.js did not match (case sensitive) the file names on disk. 

I would be looking for a better error message to enlighten idiot developers, such as myself, when we do stupid stuff like that.

**Comment by ExcelNet-Public** [comment]
@dmarkrollins:  Windows however is not case sensitive, so it should be able to handle that sort of error without issue, this would be more of a problem on other file systems that are case sensitive (Linux / MacOS).

@hwilson: Tried the steps in #8589 (reset, clear temp, update) and did seem to clear up the issues above.  Of course this is not a "fix", but a work around.  Having to reset your project is really an issue if you have significant data stored with your environment for testing.  

Thanks!

**Comment by hwillson** [comment]
Glad to hear it helped @ExcelNet-Public. That fix isn't necessarily a workaround though; clearing Meteor's temp files is sometimes necessary, as certain things can become out of synch after an upgrade. It's not a perfect fix (ideally this issue would not have happened in the first place), but you should only have to do this once. Just to add - if you don't want to clear your local DB, instead of running `meteor reset` just delete your apps `.meteor/local` directory. Alternatively, consider voting on (or better yet working on 😉) #7789.

Given that this is resolved for now, I'll close it off. Please post back if the issue comes up again, and we'll be happy to re-open. Thanks!

**Comment by dmarkrollins** [comment]
I confirmed it is indeed case sensitive on windows. Until my file names on disk and defined in package.js matched I'd receive the ENOTEMPTY error.

**Comment by ecarlotti** [comment]
I tried absolutely everything listed on this and other threads about this issue, with no positive results. I'm running meteor 1.5 and the problem started when I tried to upgrade from version 1.4.3 - and after trying it for the first time, the software would not run or build with the same error even on v1.4.3, so I had no choice other than trying to solve the issue.

Looking at the code and to my own problem, I realized that all builds are failing exactly on the same point (e.g. line of code) which is `.meteor\packages\meteor-tool\<meteor-version>\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:293:9`, (again, for Windows environments - I have not tested this on any other platform).

After taking a quick look at the code on the `rimraf.js` file, I noted that there was more than one version for the actual code that deals with folders removal, and I made a simple modification to the file, in order to "try" my theory and it worked just fine.

I changed the code on line 293 of rimraf.js from the original `rmdirSync(p, options, null)` to `rmdir(p, options, null, function(er) {if (er) throw er})` which is **NOT** a full replacement of the original function but if you have no permission issues on your Windows installation, will work just fine.

That simple change - which I understand to be only a change on the strategy used to deal with folders removal on this specific platform - that was already coded in the original file has completely solved my issue. After that I was able to build, run, debug and do everything I needed.

I really hope this information helps someone, as it helped me. I don't consider this to be a definitive fix to this issue, but instead a viable workaround for those as desperate as I was...  :-)

**Comment by abernix** [comment]
@ecarlotti Glad you found a solution that works.  https://github.com/meteor/meteor/pull/8773 was actually just merged recently with a fix for what I believe your problem is (the `ENOTEMPTY` error).  This fix is out in the latest (first) 1.5.1 beta: `1.5.1-beta.0`.  You can try updating to it right now with:

    meteor update --release 1.5.1-beta.0

I'd be very curious if this release fixes your problem!

**Comment by ecarlotti** [comment]
Jesse,

First of all, thanks for the prompt reply. What I did to validate this fix:


  1.  Completely removed Meteor, including the folders that the uninstaller would not remove
  2.  Manually reset my project by deleting the .meteor/local folder
  3.  Downloaded and installed a clean version of Meteor 1.5.0
  4.  Tried to rebuild my project – Same problem: ENOTEMPTY error

After getting to the situation where the problem occurs again (I tried to build a number of times just to be sure of recurrency) I did the update to v 1.5.1-beta.0 as per your recommendation. At the very first (and complete) build, I had the ENOTEMPTY error once, but since then I was not able to reproduce it and the software runs fine on this version.

The only problem I noticed is that sometimes I get a EPERM error on hot updates (only when a SERVER RESTART is needed) and it happens on rimraf.js:290:24

Since it seems related, on my environment Windows Defender is on and updated.

Hope this helps…
/Edson
